smart-hadoop-support/smart-hadoop/src/main/java/org/smartdata/hdfs/action/CopyFileAction.java

```diff
@@ -80,13 +100,16 @@ public void init(Map<String, String> args) {
       this.destPath = args.get(DEST_PATH);
     }
     if (args.containsKey(BUF_SIZE)) {
-      bufferSize = Integer.valueOf(args.get(BUF_SIZE));
+      bufferSize = Integer.parseInt(args.get(BUF_SIZE));
     }
     if (args.containsKey(OFFSET_INDEX)) {
-      offset = Long.valueOf(args.get(OFFSET_INDEX));
+      offset = Long.parseLong(args.get(OFFSET_INDEX));
     }
     if (args.containsKey(LENGTH)) {
-      length = Long.valueOf(args.get(LENGTH));
+      length = Long.parseLong(args.get(LENGTH));
+    }
+    if (StringUtils.isNotBlank(args.get(PRESERVE))) {
+      rawPreserveAttributes = Arrays.asList(args.get(PRESERVE).split(","));
```

Comment:
Consider getting `Set<PreserveAttribute>` here to avoid having to store `List<String> rawPreserveAttributes`, which is only needed for further parsing.